### PR TITLE
Debounce step: allows for debouncing event triggers for getting only end sequence triggers

### PR DESCRIPTION
**Pull request by Peterburnett**

### CODE REVIEW

File: classes/event_processor.php [hunk 1]
```diff
@@ -216,6 +216,13 @@ private function process_realtime_workflow($workflow, $evententry) {
             foreach ($steps as $step) {
                 try {
                     $outertransaction = $DB->is_transaction_started();
+
+                    // Special case, the debounce step should never process realtime.
+                    // Bail here, and move to the cron queue.
+                    if ($step->stepclass === '\tool_trigger\steps\debounce\debounce_step') {
```
Comment (brendanheywood) [inline classes/event_processor.php:222]: could this be inside the execute function of the step?

Comment (Peterburnett) [reply classes/event_processor.php:222]: Yeah, the identification can be based on the cron user, so we know the env

Comment (Peterburnett) [reply classes/event_processor.php]: Or we could do the even smarter thing and use CLI_SCRIPT -_- . Fixed :)

File: classes/event_processor.php [hunk 2]
```diff
@@ -216,6 +216,7 @@ private function process_realtime_workflow($workflow, $evententry) {
             foreach ($steps as $step) {
                 try {
                     $outertransaction = $DB->is_transaction_started();
+
```
Comment (brendanheywood) [inline classes/event_processor.php:219]: empty whitespace

**Pull request merged**